File: 349-intersection-of-two-arrays/intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> intersection = new HashSet<>();

        for(int i = 0; i < nums1.length; i++) {
            for(int j = 0; j < nums2.length; j++) {
                if(nums1[i] == nums2[j]) {
                    intersection.add(nums1[i]);
                }
            }
        }

        return convertSetToArray(intersection);
    }

    public int[] convertSetToArray(Set<Integer> set) {
        int index = 0;
        int[] ret = new int[set.size()];
        for(int a : set) {
            ret[index++] = a;
        }

        return ret;
    }
}

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        # sort the two arrays
        # initialize the two pointers together and 

        nums1 = sorted(nums1)
        nums2 = sorted(nums2)

        pointer1 = 0
        pointer2 = 0

        ret = []

        while pointer1 < len(nums1) and pointer2 < len(nums2):
            if nums2[pointer2] < nums1[pointer1]:
                while pointer2 < len(nums2) and nums2[pointer2] < nums1[pointer1]:
                    pointer2 += 1
            
            if pointer1 < len(nums1) and pointer2 < len(nums2) and nums2[pointer2] == nums1[pointer1] and nums1[pointer1] not in ret:
                ret.append(nums1[pointer1])
            
            pointer1 += 1
        
        return ret
